tests/integration/test_api/test_config/test_behind_proxy_server/test_behind_proxy_server.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2015-2020, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
+import time
+from secrets import token_hex
+
+import pytest
+import requests
+
+from wazuh_testing.tools.configuration import check_apply_test
+from wazuh_testing.tools.configuration import get_api_conf
+
+# Marks
+
+pytestmark = pytest.mark.server
+
+
+# Configurations
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'conf.yaml')
+configuration = get_api_conf(configurations_path)
+
+
+# Fixtures
+
+@pytest.fixture(scope='module', params=configuration)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+# Tests
+
+@pytest.mark.parametrize('tags_to_apply', [
+    {'bps_enabled'},
+    {'bps_disabled'},
+])
+@pytest.mark.parametrize('insert_agent', [
+    False,
+    True,
+])
+def test_behind_proxy_server(insert_agent, tags_to_apply, get_configuration, configure_api_environment,
+                             restart_api, wait_for_start, get_api_details):
+    """State whether the behind_proxy_server option is working.
+
+    Verify if the IP established in the X-Forwarded-For header of the request is
+    used as the agent IP when the behind_proxy_server option is enabled.
+    Otherwise, the registration IP should be different.
+
+    The POST /agents and /agents/insert methods make use of this configuration,
+    so both are tested in this  test.
+
+    Parameters
+    ----------
+    insert_agent : bool
+        Whether the endpoint to test is /agents/insert.
+    tags_to_apply : set
+        Run test if match with a configuration identifier, skip otherwise.
+    """
+    check_apply_test(tags_to_apply, get_configuration['tags'])
+    behind_proxy_server = get_configuration['configuration']['behind_proxy_server']
+    api_details = get_api_details()
+
+    # Generate random IP and name.
+    proxy_ip = '12.3.45.67'
+    data = {'name': 'test_agent'}
+
+    # Extra data is needed when the endpoint /insert is used
+    if insert_agent:
+        data['id'] = '200'
+        data['key'] = token_hex(32)
+
+    # Add X-Forwarded-For.
+    api_details['auth_headers']['X-Forwarded-For'] = proxy_ip
+    api_details['base_url'] += '/agents'
+
+    # Delete previous agents to avoid duplicate data error.
```

Comment:
I think we should avoid making the test dependent on this stage. I don't like the possibility of the config test failing if this goes wrong. Maybe use new agents.

Reply:
Now the IP and ID are also randomly generated, so even if deleting existing agents fails, it would still be difficult for the test to fail due to duplicate data.